docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "try: \n",
+    "    from netflix_movies import MovieView, MovieTitle\n",
+    "except ImportError:\n",
+    "    print(\"netflix_movies module cannot be imported. Please check!\")\n",
+    "else:\n",
+    "    print(\"MovieView and MovieTitle dataclasses have been successfully imported.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### **`S1.2`** `beam.ParDo` handler\n",
+    "\n",
+    "The [`beam.ParDo`](https://beam.apache.org/releases/pydoc/2.28.0/apache_beam.transforms.core.html#apache_beam.transforms.core.ParDo) processing paradigm is very similar to the `Map` phase of a `Map`/`Shuffle`/`Reduce`-style algorithm. \n",
+    "\n",
+    "The [`ParDo`](https://beam.apache.org/documentation/transforms/python/elementwise/pardo/) transformer is executed in parallel elementwise to the `PCollection`, via custom [`beam.DoFn`](https://beam.apache.org/releases/pydoc/2.28.0/apache_beam.transforms.core.html#apache_beam.transforms.core.DoFn) callable."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@beam.typehints.with_output_types(MovieView)  # type-hint annotation for the output\n",
+    "class ParseMovieViews(beam.DoFn):\n",
+    "    \"\"\"ParDo main execution core. Each line of the DATA_FILE will be processed, \n",
+    "    and an instance of MovieView will be created and returned, accordingly.\"\"\"\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        self.movie_id: int = -1\n",
+    "    \n",
+    "    def process(self, line: str):\n",
+    "        from netflix_movies import MovieView\n",
+    "        line = line.strip()  # get rid of any useless tabulation\n",
+    "        if not line:\n",
+    "            return  # Skip Line\n",
+    "        if line.endswith(\":\"):\n",
+    "            movie_id, _ = line.split(\":\")\n",
+    "            self.movie_id = int(movie_id)\n",
+    "            return  # Automatically skip this line\n",
+    "        user_id, rating, date_stamp, *_ = line.split(',')\n",
+    "        user_id, rating = int(user_id), int(rating)\n",
+    "        yield MovieView(user_id, self.movie_id, rating, date_stamp)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `ParseMovieViews` callable will be used to generate the initial reference `PCollection` of `MovieView` instances for the `Netflix-prize` dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now work out a solution in a very similar fashion to parse _movie_titles_ as extracted from the `movie_titles.txt` dataset: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@beam.typehints.with_output_types(MovieTitle)  # type-hint annotation for the output\n",
+    "class ParseMovieTitles(beam.DoFn):\n",
```

Comment:
Could you please add
#@title ....
and hide the code by default (it might be too much information for the user)